openassessment/fileupload/api.py

```diff
@@ -29,3 +31,160 @@ def remove_file(key):
     Remove file from the storage
     """
     return backends.get_backend().remove_file(key)
+
+
+def _safe_load_json_list(field):
+    """
+    Tries to load JSON-ified string,
+    returns an empty list if we try to load some non-JSON-encoded string.
+    """
+    try:
+        return json.loads(field)
+    except ValueError:
+        return []
+
+
+class FileUpload(object):
+    """
+    A layer of abstraction over the various components of file
+    data stored as ORA XBlock user-scoped fields.
+    """
+    def __init__(self, name, description, size, index, **student_item_dict):
+        self.name = name
+        self.description = description
+        self.size = size
+        self.index = index
+        self.student_id = student_item_dict.get('student_id')
+        self.course_id = student_item_dict.get('course_id')
+        self.item_id = student_item_dict.get('item_id')
+
+    @property
+    def exists(self):
+        return self.key is not None
+
+    @property
+    def download_url(self):
+        if self.exists:
+            return get_download_url(self.key)
```

Comment:
I know this POC - but what happens if there is no key - i.e: should we return a NULL/empty string?